EIPS/eip-7701.md

```diff
@@ -194,6 +240,49 @@ abi.encode(uint256 actualGasCost, bool success, bytes context)
 
 This step is performed with the `role_paymaster_post_tx` code section.
 
+### Transaction Execution Flow
+
+All existing transaction types only have an implicit validation phase where balance, nonce, and signature are checked,
+and an implicit execution phase with a single top-level execution frame.
+
+For all existing transaction types, during the single top-level execution frame,
+the `ORIGIN` (`0x32`, `tx.origin`) and `CALLER` (`0x33`, `msg.sender`)
+are both equal the address that is determined by the transaction's ECDSA signature (`yParity`, `r`, `s`).
+
+When processing an EIP-7701 transaction, however, multiple execution frames will be created.
+The full list of possible frames and their corresponding code sections is as follows:
+
+1. **Validation Phase**
+   * `sender` deployment frame (once per account) - `role_sender_deployment`
+   * `sender` validation frame (required) - `role_sender_validation`
+   * `paymaster` validation frame (optional) - `role_paymaster_validation`
+2. Execution Phase
+   * `sender` execution frame (required) - `role_sender_execution`
+   * `paymaster` post-transaction frame (optional) - `role_paymaster_post_tx`
+
+All execution frames in the **Validation Phase** must be completed successfully without reverting,
+and both `sender` and `paymaster` validation frames must include a call to a corresponding `AA_ENTRY_POINT`
+approval callback functions in order for the transaction to be considered valid for a given position in a block.
+
+In all top-level frames, the global variables have the following meaning:
+
+| Opcode Name | Solidity Equivalent | Value                                                                         |
+|-------------|---------------------|-------------------------------------------------------------------------------|
+| `CALLER`    | `msg.sender`        | The `AA_ENTRY_POINT` address. `AA_SENDER_CREATOR` for the "deployment frame". |
+| `ORIGIN`    | `tx.origin`         | The transaction `sender` address                                              |
+| `CALLDATA*` | `msg.data`          | The transaction data is set to inputs of the corresponding frame              |
+
+### Transaction execution context
```

Comment:
This entire section seems to belong in the "Backwards Compatibility" section that EIPs typically have. It basically says that these opcodes behave the same way they'd behave in a legacy transaction despite having multiple call frames.